report/report.md

```diff
@@ -0,0 +1,898 @@
+
+- $\IVCProver(z_i: \textbf{Instance}, \pi_i: \textbf{Proof}, acc_i: \Acc) \to (\textbf{Proof}, \Acc)$
+
+  - Accumulates $(z_i, \pi_i)$ with $acc_i$ to obtain a new accumulator $acc_{i+1}$.
+  - Then generates a SNARK proof $\pi_{i+1}$ of the following claim, expressed as a circuit $R$:
+      $$"z_{i+1} = F(z_i) \; \land \; (i = 0 \lor \exists \, \pi_i, \, acc_i \text{ s.t. } \ASVerifier((z_i, \pi_i), acc_i, acc_{i+1}) = \top)"$$
+  - Output $(\pi_{i+1}, acc_{i+1})$
+
+- $\IVCVerifier(\pi_{i+1}: \textbf{Proof}, acc_{i+1}: \Acc) \to \Result(\top, \bot)$
+
+  Checks the proof: $\top \meq \SNARKVerifierSlow(\pi_{i+1}) \meq \ASDecider(acc_{i+1})$
+
+Consider the above chain run $n$ times. As in the "simple" SNARK IVC
+construction, if the SNARK verifier accepts at the end, then that means:
+
+$$
+\begin{alignedat}[b]{2}
+  &\SNARKVerifierSlow(\pi_i)                               &&= \top \then \\
+  &\ASVerifier((z_{n-1}, \pi_{n-1}), acc_{n-1}, acc_n)     &&= \top \then \\
+  &\ASVerifier((z_{n-2}, \pi_{n-2}), acc_{n-2}, acc_{n-1}) &&= \top \then \cdots \\
+  &\ASVerifier((z_0, \pi_0), acc_0, acc_1)                 &&= \top \then \\
+\end{alignedat}
+$$
+
+Now, by the soundness property of the Accumulation Scheme, and instance
+$q \in X$ will be true if $\top = \ASVerifier(q, acc_i, acc_{i+1}) =
+\ASDecider(acc_{i+1})$, so if all the accumulators $\vec{acc} \in
+\Acc^{n+1}$ are valid, then all the instances will be true. This is exactly
+the case however due to the definition of the decider whereby checking an
+accumulator $acc_i$ ensures that every previous instance is true, $\Phi(q_i)
+= \top$, provided that all previous verifiers accepted.
+
+### General Polynomial Commitment Schemes
+
+In the section above about SNARKs, general-purpose proof schemes were
+described. Modern general-purpose (zero-knowledge) proof schemes, such as
+Sonic[^1], Plonk[^2] and Marlin[^3], commonly use PCS's _Polynomial Commitment
+Schemes_ for creating their proofs. This means that different PCS's can be
+used to get security under weaker or stronger assumptions.
+
+<!-- AGM gives you snarks? -->
+**TODO**: List the options (AGM?, BP, STARKS).
+
+- **AGM:** Uses a trusted setup, this would give you a traditional SNARK.
+- **DL:** Uses an untrusted setup, assumed secure if the Discrete Log problem is hard, the verifier is linear.
+- **FRI:** Also uses an untrusted setup, assumed secure one way functions
+  exist. It has a higher overhead than PCSs based on the Discrete Log assumption.
+
+**TODO**: The assumptions on $d$ seems wrong in the code, not a big deal
+
+A PCS allows a prover to prove to a verifier that a commited polynomial
+evaluates to a certain value, $v$, given an evaluation input $z$.There are
+three main functions used to prove this ($\PCSetup$ and $\PCTrim$ omitted):
+
+- $\PCCommit(p: \Fb^{d'}_q[X], d: \Nb, \o: \Option(\Fb_q)) \to \Eb(\Fb_q)$
+
+  Commits to a polynomial $p$ with degree bound $d$ where $d \geq d'$ using
```

Comment:
What is d'? Degree of p? Then we can perhaps say something like "...a degree-d' polynomial $p$..."?

Reply:
Yes exactly, it's specified in the `\Fb^{d'}_q[X]` but I like the clarification suggested.